### PR TITLE
Add feature flag for removing detailed error from hash calcution

### PR DESCRIPTION
**Pull request by areshand**
### Description
reserve a feature flag to enable new txn output

### CODE REVIEW

**Review by msmouse — COMMENTED** [review]
inb4 compilation failure 😀 

(you need to add it to aptos-release-builder::components::feature_flags as well and the conversions)

File: types/src/on_chain_config/aptos_features.rs [hunk 1]
```diff
@@ -106,6 +107,9 @@ impl Features {
         byte_index < self.features.len() && (self.features[byte_index] & bit_mask != 0)
     }
 
+    pub fn is_remove_detailed_error_from_hash_enabled(&self) -> bool {
+        self.is_enabled(FeatureFlag::REMOVE_DETAILED_ERROR_FROM_HASH)
+    }
```
Comment (msmouse) [inline types/src/on_chain_config/aptos_features.rs:112]: blank line below. 

why not append to the end?

**Review by msmouse — COMMENTED** [review]
need to build and submit framework documentation changes I guess?

File: types/src/on_chain_config/aptos_features.rs (hunk 1 above)
Comment (areshand) [reply types/src/on_chain_config/aptos_features.rs:112]: not sure if the order matters?

**Comment by areshand** [comment]
> need to build and submit framework documentation changes I guess?

what is this "framework documentation"? is there a link?

**Comment by msmouse** [comment]
> what is this "framework documentation"? is there a link?

lint, and you'll see (the changes in the .md files)

**Review by msmouse — APPROVED** [review]

File: types/src/on_chain_config/aptos_features.rs (hunk 1 above)
Comment (msmouse) [reply types/src/on_chain_config/aptos_features.rs]: it doesn't matter (much). but the question is "why not?".

Comment (areshand) [reply types/src/on_chain_config/aptos_features.rs]: already move to the end

**Pull request merged**